src/input/composite_device/mod.rs

```diff
@@ -231,22 +236,25 @@ impl CompositeDevice {
         Ok(device)
     }
 
+    /// Return the DBus path of the composite device
+    pub fn dbus_path(&self) -> String {
+        self.dbus_path.clone()
+    }
```

Comment:
Maybe we should return the `&str` and let the caller decide if they want to `clone()`.

```suggestion
    pub fn dbus_path(&self) -> &str {
        self.dbus_path.as_str()
    }
```

Reply:
Done